graphqlapi/src/main/java/io/stargate/graphql/schema/graphqlfirst/migration/CreateTableQuery.java

```diff
@@ -62,6 +68,18 @@ public String getDescription() {
     return "Create table " + table.name();
   }
 
+  @Override
+  public void authorize(AuthorizationService authorizationService, AuthenticationSubject subject)
+      throws UnauthorizedException {
+    authorizationService.authorizeSchemaWrite(
+        subject,
+        table.keyspace(),
+        table.name(),
+        Scope.CREATE,
+        SourceAPI.GRAPHQL,
+        ResourceKind.TABLE);
```

Comment:
Not sure here either. It feels wrong to reference a table that does not exist yet.

It looks like [Cassandra](https://github.com/apache/cassandra/blob/trunk/src/java/org/apache/cassandra/cql3/statements/schema/CreateTableStatement.java#L128) checks the `CREATE` permission on the keyspace. Should I leave the table name null?

Reply:
Current code matches the auth logic in `StargateQueryHandler` :+1: